service/pipeline.ml

```diff
@@ -43,9 +43,28 @@ let job_id x =
   let+ job = Current.Analysis.get x in
   Current.Analysis.job_id job
 
+module Docker_of_builder (Builder : Conf.BUILDER) : Ocaml_ci.S.DOCKER_CONTEXT = struct
+
+  type image = Builder.Image.t
+
+  let image_hash = Builder.Image.hash
+
+  let pull ~schedule name =
+    Builder.pull ~schedule name
+
+  let build ~label ~dockerfile source =
+    Builder.build ~label ~pool:Builder.pool ~pull:false ~dockerfile source
```

Comment:
I don't see why you specialised `~pull:false` here, even if that's the only value that we happen to use right now.

Reply:
Does `~pull:true` makes sense when building from a Dockerfile ? If that is useful in the future, would a separate function makes sense ? Or at least an optional argument with the right default ?

Reply:
I don't think the use of a Dockerfile or otherwise really matters; `--pull` updates the base image specified in the Dockerfile.

Reply:
What matters is whether the Dockerfile specifies a tag (which might change) or a hash (which won't). I think ocaml-ci always uses hashes in the Dockerfiles, so `pull` doesn't do anything.